conformance/utils/suite/suite.go

```diff
@@ -77,6 +77,23 @@ var StandardCoreFeatures = map[SupportedFeature]bool{
 	SupportReferenceGrant: true,
 }
 
+// AllFeatures contains all the supported features and can be used to run all
+// conformance tests with `all-features` flag.
+var AllFeatures = map[SupportedFeature]bool{
```

Comment:
IMO, there is no native set structure in golang, do you mean the `map[SupportedFeature]struct{}`? Alsowe should pass `AllFeatures` to suite's `SupportedFeatures`, so we should make them the same type.

Reply:
Sets are available in apimachinery:

https://github.com/kubernetes/apimachinery/tree/master/pkg/util/sets

e.g.:

```go
import "k8s.io/apimachinery/pkg/utils/sets"

var AllFeatures = sets.NewString(
	SupportReferenceGrant,
	SupportTLSRoute,
	SupportHTTPRouteQueryParamMatching,
	SupportHTTPRouteMethodMatching,
	SupportHTTPResponseHeaderModification,
	SupportRouteDestinationPortMatching,
	SupportGatewayClassObservedGenerationBump,
	SupportHTTPRoutePortRedirect,
	SupportHTTPRouteSchemeRedirect,
	SupportHTTPRoutePathRedirect,
	SupportHTTPRouteHostRewrite,
	SupportHTTPRoutePathRewrite,
)
```

```go
if AllFeatures.Has(SupportReferenceGrant) {
```

Reply:
Yes the implementation ultimately uses a map, so the benefit here is for the developers which I think is a fair thing to trade for, however I don't consider this suggestion a blocker I'll let others weigh in.

Reply:
Understood, if others also want the benefit, I can implement it.

Reply:
Now that the k8s sets util supports generics, I think it would be best to move both this and AllFeatures over to use that approach.

Reply:
> Now that the k8s sets util supports generics, I think it would be best to move both this and AllFeatures over to use that approach.

@robscott @shaneutt I tried to move it to K8s sets utils: https://github.com/kubernetes-sigs/gateway-api/pull/1642/commits/b85174a868892238ff589643da738f492fda4e35, can you review if it is correct?